clang/include/clang/Driver/Options.td

```diff
@@ -2982,6 +2982,10 @@ def fclangir_call_conv_lowering : Flag<["-"], "fclangir-call-conv-lowering">,
   Visibility<[ClangOption, CC1Option]>, Group<f_Group>,
   HelpText<"Enable ClangIR calling convention lowering">,
   MarshallingInfoFlag<FrontendOpts<"ClangIREnableCallConvLowering">>;
+def fclangir_mem2reg : Flag<["-"], "fclangir-mem2reg">,
```

Comment:
Might be a good a idea to do the `defm` version already as to populate the "no" variant, the idea is to make it handy to disable whenever we flip this by default while testing in our codebase (or when anyone else tries to do the same).